libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -656,6 +656,15 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		solAssert(_typeOnStack == _targetType, "Invalid type conversion requested.");
 		break;
 	}
+
+	// Check the conversion result fits in a range.
+	if (targetTypeCategory == Type::Category::Enum)
```

Comment:
In that case, I would call the cleaning routine at this spot.  This catches anything -> enum conversion.